scripts/globals/fishing_contest.lua

```diff
@@ -0,0 +1,417 @@
+local function isRewardAvailable(player)
+    local awards = player:getContestRewardStatus()
+
+    if awards ~= nil then
+        if
+            awards['rank'] > 0 and
+            awards['rank'] <= 20
+        then
+            return 1
+        end
+    end
+
+    return 0
+end
+
+-- This returns the data necessary to send the packet with the 'time remaining until stage change' data to the player
+local function getTimeRemaining(changeTime)
+    local timeTable = {}
+    local currentTime = os.time()
+    if changeTime > currentTime then
+        local timeDelta = changeTime - currentTime
+        timeTable =
+        {
+            ['days']    = math.floor(timeDelta / (60 * 60 * 24)),
+            ['hours']   = math.floor((timeDelta / (60 * 60)) % 24),
+            ['minutes'] = math.floor((timeDelta / 60) % 60),
+        }
+    else
+        timeTable =
+        {
+            ['days']    = 0,
+            ['hours']   = 0,
+            ['minutes'] = 0,
+        }
+    end
+
+    return timeTable
+end
+
+-- Provides the score of the submission
+local function scoreFish(length, weight, criteria)
+    local score = 0
+    if criteria == xi.fishingcontest.criteria.SIZE then
+        score = length
+    elseif criteria == xi.fishingcontest.criteria.WEIGHT then
+        score = weight
+    else
+        score = length + weight
+    end
+
+    return score
+end
+
+-- Determines the rewards to be granted based on rank and share
+local function getPlayerReward(player)
+    local status = player:getContestRewardStatus()
+
+    if status ~= nil then
+        local rank  = status['rank']
+        local share = status['share'] or 1
+        if
+            rank > 0 and
+            rank <= 20
+        then
+            local reward = xi.fishingcontest.reward[rank]
+            if reward.gil then
+                reward.gil = reward.gil / share
+            end
+
+            return reward
+        end
+    end
+
+    return nil
+end
+
+-----------------------------------
+-- HOOKED FUNCTIONS
+-----------------------------------
+xi.fishingcontest.selectContestFish = function()
+    local fishList = xi.fishingcontest.fish
+    return xi.fishingcontest.fish[math.random(#fishList)]['id']
+end
+
+xi.fishingcontest.getStageChangeTime = function(stage, stageStartTime)
+    -- If the value is invalid, send back the 'never change' time
+    if
+        stage >= xi.fishingcontest.status.CONTESTING and
+        stage < xi.fishingcontest.status.CLOSED
+    then
+        return stageStartTime + xi.fishingcontest.interval[stage]
+    end
+
+    return xi.fishingcontest.interval[xi.fishingcontest.status.CLOSED]
+end
+
+xi.fishingcontest.onStageProgress = function(newStage)
+    -- Place holder -- I believe there is an announcement made by the NPC at certain stage progressions
+end
+
+xi.fishingcontest.onTrigger = function(player, npc)
+    local contest = GetFishingContest()
+
+    player:startEvent(10006, {
```

Comment:
Sneaky sneaky, I totally forgot startEvent() could take a table in it's sol::variadic_args